Lib/tokenize.py

```diff
@@ -492,8 +492,14 @@ def _tokenize(readline, encoding):
             # BOM will already have been stripped.
             encoding = "utf-8"
         yield TokenInfo(ENCODING, encoding, (0, 0), (0, 0), '')
+    last_line = b''
+    line = b''
     while True:                                # loop over lines in stream
         try:
+            # This loop has multiple points where it can break out so we
+            # pick up the value for the last_line here, at one unified
+            # point to keep things simple.
```

Comment:
"at one unified point to keep things simple" should be removed IMO (unnecessary).

I suggest mentioning briefly *why* `last_line` is kept (to the best of my understanding: the last line is needed for the newline check after the loop, but `line` will be overridden at the last loop iteration).